diff --git a/includes/SpecialContact.php b/includes/SpecialContact.php
index 5fc97e4..34d379a 100644
--- a/includes/SpecialContact.php
+++ b/includes/SpecialContact.php
@@ -298,6 +298,7 @@ class SpecialContact extends UnlistedSpecialPage {
 			if ( $this->getConfig()->get( 'UserEmailUseReplyTo' ) ) {
 				// Define reply-to address
 				$replyTo = $senderAddress;
+				$senderAddress = $contactSender;
 			}
 		}
 
@@ -437,6 +438,8 @@ class SpecialContact extends UnlistedSpecialPage {
 		// if the user requested a copy of this mail, do this now,
 		// unless they are emailing themselves, in which case one copy of the message is sufficient.
 		if ( $formData['CCme'] && $fromAddress ) {
+			// We recreate it in case $wgUserEmailUseReplyTo was used.
+			$senderAddress = new MailAddress( $fromAddress, $fromName );
 			$cc_subject = $this->msg( 'emailccsubject', $contactRecipientUser->getName(), $subject )->text();
 			if ( Hooks::run( 'ContactForm',
 				[ &$senderAddress, &$contactSender, &$cc_subject, &$text, $this->formType, $formData ] )
