diff --git a/Maps_Settings.php b/Maps_Settings.php
index 70ca8c2d..f04d1ab1 100644
--- a/Maps_Settings.php
+++ b/Maps_Settings.php
@@ -63,6 +63,9 @@
 
 	// Boolean. Sets if geocoded addresses should be stored in a cache.
 	$GLOBALS['egMapsEnableGeoCache'] = true;
+	// Integer. If egMapsEnableGeoCache is true, determines the TTL of cached geocoded addresses.
+	// Default value: 1 day.
+	$GLOBALS['egMapsGeoCacheTtl'] = BagOStuff::TTL_DAY;
 
 
 // Coordinate configuration
diff --git a/src/Geocoders/CachingGeocoder.php b/src/Geocoders/CachingGeocoder.php
index 4e8fdb5f..1b70f44d 100644
--- a/src/Geocoders/CachingGeocoder.php
+++ b/src/Geocoders/CachingGeocoder.php
@@ -16,10 +16,12 @@ class CachingGeocoder implements Geocoder {
 
 	private $geocoder;
 	private $cache;
+	private $cacheTtl;
 
-	public function __construct( Geocoder $geocoder, BagOStuff $cache ) {
+	public function __construct( Geocoder $geocoder, BagOStuff $cache, $cacheTtl ) {
 		$this->geocoder = $geocoder;
 		$this->cache = $cache;
+		$this->cacheTtl = $cacheTtl;
 	}
 
 	/**
@@ -34,7 +36,7 @@ public function geocode( string $address ) {
 		if ( $coordinates === false ) {
 			$coordinates = $this->geocoder->geocode( $address );
 
-			$this->cache->set( $key, $coordinates, BagOStuff::TTL_DAY );
+			$this->cache->set( $key, $coordinates, $this->cacheTtl );
 		}
 
 		return $coordinates;
diff --git a/src/MapsFactory.php b/src/MapsFactory.php
index 1723a508..d54ea756 100644
--- a/src/MapsFactory.php
+++ b/src/MapsFactory.php
@@ -37,7 +37,8 @@ public function newGeocoder(): Geocoder {
 		if ( $this->settings['egMapsEnableGeoCache'] ) {
 			return new CachingGeocoder(
 				$geocoder,
-				$this->getMediaWikiCache()
+				$this->getMediaWikiCache(),
+				$this->settings['egMapsGeoCacheTtl']
 			);
 		}
 

