diff --git a/src/SemanticFormsSelectInput.php b/src/SemanticFormsSelectInput.php
index 9043161..7d3dab0 100644
--- a/src/SemanticFormsSelectInput.php
+++ b/src/SemanticFormsSelectInput.php
@@ -248,9 +248,9 @@ class SemanticFormsSelectInput extends PFFormInput {
 
 				} else {
 					if ( in_array( $cur, $curvalues ) ) {
-						$selected = " selected='selected'";
+						$selected = ['selected' => TRUE];
 					}
-					$ret .= "<option" . $selected . ">$cur</option>";
+					$ret .= \Html::element('option', $selected, $cur);
 				}
 			}
 
