diff --git a/src/Query/ResultPrinters/CsvFileExportPrinter.php b/src/Query/ResultPrinters/CsvFileExportPrinter.php
index 29e335755..9124bbd49 100644
--- a/src/Query/ResultPrinters/CsvFileExportPrinter.php
+++ b/src/Query/ResultPrinters/CsvFileExportPrinter.php
@@ -157,7 +157,7 @@ class CsvFileExportPrinter extends FileExportPrinter {
 				$growing = [];
 
 				while ( ( $object = $field->getNextDataValue() ) !== false ) {
-					$growing[] = Sanitizer::decodeCharReferences( $object->getShortWikiText() );
+					$growing[] = addcslashes( Sanitizer::decodeCharReferences( $object->getShortWikiText() ), ',' );
 				}
 
 				$row_items[] = implode( $vsep, $growing );
